### ISSUE TITLE
Question on Proxy-NCA

### ISSUE THREAD

**Issue by WeidiXie** [Confusezius/Deep-Metric-Learning-Baselines#17]
Hi, thanks for the nice code. I noticed that your re-implementation of the Proxy-NCA is much higher than that from their original paper on CUB-200, e.g. R@1= 64.0 vs R@1= 49.

I understand that the original paper uses Inception, and you train a ResNet50, are the performance boost all from architecture ? Or is there any other implementation differences ?

**Comment by Confusezius** [comment]
So I generally found that under the same implementation ProxyNCA performs similar to other Sota methods, with super fast convergence speed. There are some hyperparameters that required a bit of tweaking. I especially found that increasing the distance of proxies to anchor during training (by multiplying each element with a factor in [2,4]) helped convergence and performance. If you remove the factor of 3 (lines 494,495) you should see what I mean. Inspiration for this I took from https://github.com/dichotomies/proxy-nca/blob/master/proxynca.py who noticed the same thing.

**Comment by WeidiXie** [comment]
I see, thanks, I always think the proxyNCA can be quite powerful, but the original paper shows very marginal improvements.

I guess it might be related to the temperature in the softmax function, I saw people use this in the Large-margin softmax, AMSoftmax, etc. As long as you normalise the features, squashes all the features to a hypersphere, the gradients tend to also be squashed, which may make the training very difficult.

**Comment by Confusezius** [comment]
That's a nice pointer! The relation to temperature seems quite sensible :).
